tests/reflection/advanced-features.ts

```diff
@@ -0,0 +1,12 @@
+// automatically generated by the FlatBuffers compiler, do not modify
```

Comment:
should this and all other related files be in `reflection` rather than `tests` ?

Reply:
do we have TS code that relies on reflection?

Reply:
this does indeed seem confused between tests/reflection. I don't know anything about it and note that is was also added by #7161.